File: code_combat/7_Sarven_Desert/325-Marauder/marauder.js
// Destroy mechs and collect gold from them.

while (true) {
    var coin = hero.findNearestItem();
    // While a coin exists:
    while (coin) {
        // Move to the coin.
        hero.moveXY(coin.pos.x, coin.pos.y);
        // Reassign the coin variable to the nearest item.
        coin = hero.findNearestItem();
    }
    var enemy = hero.findNearestEnemy();
    if (enemy) {
        // While enemy's health is greater than 0.
        while (enemy.health > 0) {
            // Attack enemy.
            if (hero.isReady("bash")) hero.bash(enemy);
            else hero.attack(enemy);
        }
    }
}

File: code_combat/7_Sarven_Desert/325-Marauder/marauder.py
while True:
    coin = hero.findNearestItem()
    while coin:
        hero.moveXY(coin.pos.x, coin.pos.y)
        coin = hero.findNearestItem()

    enemy = hero.findNearestEnemy()
    if enemy:
        while enemy.health > 0:
            if hero.isReady("bash"):
                hero.bash(enemy)
            else:
                hero.attack(enemy)
